src/main/java/com/base2services/jenkins/github/GitHubTriggerProcessor.java

```diff
@@ -71,22 +102,117 @@ public void processGitHubPayload(String payload, Class<? extends Trigger> trigge
         }
     }
 
-    private JSONObject extractJsonFromPayload(String payload) {
-        JSONObject repository = null;
+    public JSONObject extractJsonFromPayload(String payload) {
```

Comment:
Reviewer note:
Changed this function to public so that it can be used in tests.
Also changed its functionality to be more generic - as written it would not extract the payload if there was no "repository" object in it but instead would return null.